### PR TITLE
decode: reject strings with negative length

### PR DESCRIPTION
**Pull request by pmezard**
Found running go-fuzz.

### CODE REVIEW

**Comment by zeebo** [comment]
Nice! Would you mind adding yourself to AUTHORS? Also do you agree to license this contribution under the terms in the LICENSE file?

File: decode_test.go [hunk 1]
```diff
@@ -79,6 +79,7 @@ func TestDecode(t *testing.T) {
 		{`di5ei2ee`, new(interface{}), nil, true},
 		{`d3:fooe`, new(interface{}), nil, true},
 		{`l3:foo3:bar`, new(interface{}), nil, true},
+		{`d3:000d3:000ld-6:`, new(interface{}), nil, true},
```
Comment (zeebo) [inline decode_test.go]: also i believe `d-1` is a shorter test case that exhibits the same bug.

**Comment by pmezard** [comment]
PTAL

**Comment by zeebo** [comment]
Thanks!

**Pull request merged**